File: csharp/EncodeKata/ByteBuffer.cs
using System.Collections.Generic;

namespace EncodeKata
{
    public class ByteBuffer
    {
        private List<int> buffer = new List<int>();
        private int current;
        public void Append(int i)
        {
            buffer.Add(i);
        }

        public int Read()
        {
            var value = buffer[current];
            current += 1;
            return value;
        }

        public int GetAvailable()
        {
            return buffer.Count - current;
        }
    }
}

File: java/src/main/java/encodeKata/ByteBuffer.java
package encodeKata;

import java.util.ArrayList;
import java.util.List;

public class ByteBuffer {
    private final List<Integer> buffer = new ArrayList<>();
    private int current;

    public void append(int i) {
        buffer.add(i);
    }

    public int read() {
        int value = buffer.get(current);
        current += 1;
        return value;
    }

    public int getAvailable() {
        return buffer.size() - current;
    }

}